lib/Tapable.js

```diff
@@ -2,6 +2,42 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
+
+// polyfill from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
+// using the polyfill specifically to avoid the call to `Object.defineProperty` for performance reasons
+function fastFilter(fun/*, thisArg*/) {
```

Comment:
Could you rewrite this to the minimal needed filter function? It don't have to be a perfect polyfill.

``` js
function fastFilter(array, fun) {
  "use strict";
  var res = [];
  for(var i = 0; i < array.length; i++) {
    var val = array[i];
    if(func(val)) res.push(val);
  }
  return res;
}
```

and check the performance on this one.